src/Attributes/Route.php

```diff
@@ -73,6 +73,24 @@ function (array $params) {
                 array_walk($params, fn(Parameter $param) => $this->getParams[] = $param);
             }
         );
+
+        if (preg_match_all('#{([^}]+)}#', $this->route, $matches)) {
+            $pathParms = array_combine($matches[1], $matches[1]);
+            foreach ($getParams as $getParam) {
+                $paramName = $getParam->getName();
+                if (isset($pathParms[$paramName])) {
+                    unset($pathParms[$paramName]);
+                }
+            }
+            if ($pathParms) {
+                foreach ($pathParms as $pathParm) {
+                    $param = new Parameter();
+                    $param->setName($pathParm);
+                    $param->setParamType('string');
+                    $this->getParams[] = $param;
+                }
+            }
+        }
```

Comment:
I tried your code and it didn't work. This is my proposition:
```php
if (preg_match_all('#{([^}]+)}#', $this->route, $matches)) {
    $pathParams = $matches[1];
    $declaredParamsName = array_map(fn(Parameter $parameter): string => $parameter->getName(), ...$getParams);
    $unresolvedParams = array_diff($pathParams, $declaredParamsName);

    foreach ($unresolvedParams as $pathParam) {
        $param = new Parameter();
        $param->setName($pathParam);
        $param->setParamType(Type::STRING);
        $this->getParams[] = $param;
    }
}
```

What do you think ?

Reply:
Ah, $getParams is a `Parameter[][]` not a `Parameter[]`, they are all documented as just array, so got no warning about it, and as my function had zero parameters, I just got an empty array anyway.

Reply:
Your is a little better then mine, as it works for zero or one parameters, but if I remember how array_map works with multiple array parameters, it send them as separate parameters to the callback, so `$declaredParamsName` only contains the first parameters name.

Reply:
Using your version, but with `$this->getParams` instead, seams to work.

https://github.com/uderline/openapi-php-attributes/blob/f1d45fda6083eb45e8dd7c3fa27ed7798877831f/src/Attributes/Route.php#L84

But alternative, we could flatten the params earlier, see PR #15 as an alternative.

I think I prefer #15 over this #12, but I let you decide how you want it.